src/plugins/lua/arc.lua

```diff
@@ -496,8 +496,10 @@ local function arc_sign_seal(task, params, header)
 
   task:set_milter_reply({
     add_headers = {
-      ['ARC-Authentication-Results'] = {order = 1, value = cur_auth_results},
-      ['ARC-Message-Signature'] = {order = 1, value = header},
+      ['ARC-Authentication-Results'] = {order = 1, value = lua_util.fold_header(task,
+        'ARC-Authentication-Results', cur_auth_results) },
+      ['ARC-Message-Signature'] = {order = 1, value = lua_util.fold_header(task,
```

Comment:
Header is already folded there (L460):

```lua
header = lua_util.fold_header(task,
    'ARC-Message-Signature',
    header)
```

Same for AAR:

```
cur_auth_results = lua_util.fold_header(task,
      'ARC-Authentication-Results',
      cur_auth_results, ';')
```

Reply:
I find the source of the problem. It's Gmail :-) 

I captured smtp packets between my mailserver and Gmail servers and the headers are folded properly. But Gmail's "Show original", for unknown reason, shows the headers unfolded. 

When I forward this email from Gmail to another server, the headers are folded properly too.

Sorry for the inconvenience, I am closing this PR.